File: HackerEarth/Java/BasicsProgramming/InputOutput/Anagrams.java
/*
 * Given two strings, a and b , that may or may not be of the same length, 
 * determine the minimum number of character deletions required to make a and b 
 * anagrams. Any characters can be deleted from either of the strings.
 * 
 * Input :
 * test cases,t
 * two strings a and b, for each test case
 * 
 * Output:
 * Desired O/p
 * 
 * Constraints :
 * string lengths ≤ 10000
 * 
 * Note :
 * Anagram of a word is formed by rearranging the letters of the word.
 * 
 * For e.g. -> For the word RAM - MAR,ARM,AMR,RMA etc. are few anagrams.
 */
package BasicsProgramming.InputOutput;

import java.util.*;
import java.io.*;

/**
 * @author Cristian Chitiva
 * @email cychitivav@unal.edu.co
 * @since 29/10/2017
 */
public class Anagrams {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        // All code application logic here
        File f = new File("input.txt");
        Scanner in = new Scanner(System.in);

        int T = in.nextInt();

        while (T > 0) {
            char[] A = in.next().toCharArray();
            char[] B = in.next().toCharArray();

            int equal = 0;

            for (int i = 0; i < A.length; i++) {
                for (int j = 0; j < B.length; j++) {
                    if (A[i] == B[j]) {
                        equal++;
                        B[j] = 0;
                        break;
                    }
                }
            }
            System.out.println(A.length + B.length - (2 * equal));
            T--;
        }
    }

}

File: hackerearth/basic_programming/input_output/Basics_of_input_output/Anagrams.py
#!/usr/bin/env python
"""
Given two strings, a and b , that may or may not be of the same length, determine the minimum number of character deletions required to make a and b anagrams. Any characters can be deleted from either of the strings.

INPUT:
Test cases,t.
Two strings a and b, for each test case.

OUTPUT:
Desired O/p

CONSTRAINTS:
string lengths ≤ 10000

Note:
Anagram of a word is formed by rearranging the letters of the word.
For e.g. -> For the word RAM - MAR,ARM,AMR,RMA etc. are few anagrams.
"""

__author__ = "Cristian Chitiva"
__date__ = "March 17, 2019"
__email__ = "cychitivav@unal.edu.co"


T = int(input())

while T > 0:
    A = list(input().lower())
    B = list(input().lower())
    total = len(A) + len(B)
    common = 0

    for c in A:
        if c in B:
            B.remove(c)
            common += 2

    print(total - common)
    T -= 1
